src/main/java/org/skroutz/elasticsearch/index/analysis/WordDelimiterFilter.java

```diff
@@ -246,9 +246,10 @@ public boolean incrementToken() throws IOException {
         iterator.next();
 
         // word of no delimiters, or protected word: just return it
-        if ((iterator.current == 0 && iterator.end == termLength) ||
-                (protWords != null && protWords.contains(termBuffer, 0, termLength))
-                || protectedWords.contains(termAttribute.toString())) {
+        if ((termAttribute.length() == 2 && isAlphaNumeric(termAttribute.toString())
```

Comment:
Instead of converting `termAttribute` to string, we can get the char array by calling [`buffer()`](https://lucene.apache.org/core/6_5_1/core/org/apache/lucene/analysis/tokenattributes/CharTermAttribute.html#buffer--), this way we don't need to convert again to char array in the `isAlphaNumeric` function.

Even better, we already have the buffer (char array) in the `termBuffer` variable.

Reply:
termBuffer contains a queue of a previously inserted string (something like "bogusTerm" - not sure about its purpose). Therefore I used Arrays.copyOfRange to get a [0, termLength] slice of that.